### PR TITLE
Bdog 1512 truncate update

### PR DESCRIPTION
**Pull request by colin-lamed**

### CODE REVIEW

File: src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala [hunk 1]
```diff
@@ -58,41 +61,45 @@ trait HttpAuditing {
...
+      // short-circuit the payload creation (and truncation log warning)
+      if (auditConnector.isEnabled) {
+        val httpRequest = HttpRequest(verb, url.toString, request.headers, request.body, now())
+        responseF.map {
+          response => audit(httpRequest, response)
+        }.recover {
+          case e: Throwable => auditRequestWithException(httpRequest, e.getMessage)
+        }
+      } else
+        // TODO is this needed? (Would have been logged if we hadn't short-circuited)
```
Comment (colin-lamed) [inline src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala:79]: This questions applies to the log inside [AuditConnector](https://github.com/hmrc/play-auditing/blob/master/src-common/main/scala/uk/gov/hmrc/play/audit/http/connector/AuditConnector.scala#L102) too. We get a log entry for every call when disabled. Would a single log to warn it has been disabled (without requestId and sessionId) be sufficient?

File: src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala [hunk 2]
```diff
@@ -213,13 +222,38 @@ trait HttpAuditing {
...
+// functions are reused in bootstrap-play's AuditFilter
+object AuditUtils {
+  private val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  def responseBodyToMap[A](loggingContext: String, body: Body[A])(maskFunction: A => String): Map[String, String] =
+    (body match {
+        case Body.Complete(b)  => Map(ResponseMessage     -> maskFunction(b))
+        case Body.Truncated(b) => logger.warn(s"$loggingContext response body was truncated for auditing")
+                                  Map(ResponseMessage     -> maskFunction(b),
+                                      ResponseIsTruncated -> true.toString
```
Comment (colin-lamed) [inline src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala:245]: Here are the flags sent to datastream to indicate truncation/omission

**Review by tomwadeson — APPROVED** [review]

File: src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala [hunk 3]
```diff
@@ -213,13 +222,38 @@ trait HttpAuditing {
...
 }
+
+// functions are reused in bootstrap-play's AuditFilter
+object AuditUtils {
+  private val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  def responseBodyToMap[A](loggingContext: String, body: Body[A])(maskFunction: A => String): Map[String, String] =
+    (body match {
+        case Body.Complete(b)  => Map(ResponseMessage     -> maskFunction(b))
+        case Body.Truncated(b) => logger.warn(s"$loggingContext response body was truncated for auditing")
```
Comment (colin-lamed) [inline src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala:243]: can we downgrade the log to `info`?

**Review by tomwadeson — DISMISSED** [review]

File: src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala [hunk 4]
```diff
@@ -92,7 +92,7 @@ trait HttpAuditing {
   )(implicit hc: HeaderCarrier) = {
     import AuditExtensions._
     val isRequestTruncated =
-      request.body.isTruncated
+      request.body.fold(false)(_.isTruncated)
```
Comment (tomwadeson) [inline src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala:95]: `exists`?

**Review by tomwadeson — APPROVED** [review]

**Pull request merged**